src/Common/src/Interop/Unix/libgss/Interop.GssApi.cs

```diff
@@ -0,0 +1,162 @@
+            out uint outFlags)
+        {
+            outputBuffer = null;
+            outFlags = 0;
+
+            if (context == null)
+            {
+                context = new SafeGssContextHandle();
+            }
+
+            using (SafeGssBufferHandle outputToken = new SafeGssBufferHandle())
+            {
+                libgssapi.Status status;
+                libgssapi.Status minorStatus;
+
+                status = libgssapi.GssInitSecContext(
+                                    out minorStatus,
+                                    credential,
+                                    ref context,
+                                    isNtlm,
+                                    targetName,
+                                    (uint)inFlags,
+                                    buffer,
+                                    (buffer == null) ? 0 : buffer.Length,
+                                    outputToken,
+                                    out outFlags);
+
+                if ((status != libgssapi.Status.GSS_S_COMPLETE) && (status != libgssapi.Status.GSS_S_CONTINUE_NEEDED))
+                {
+                    throw libgssapi.GssApiException.Create(SR.net_context_establishment_failed, status, minorStatus);
+                }
+
+                outputBuffer = new byte[outputToken.Length]; // Always return non-null
+                if (outputToken.Length > 0)
+                {
+                    Marshal.Copy(outputToken.Value, outputBuffer, 0, outputToken.Length);
+                }
+                return (status == libgssapi.Status.GSS_S_COMPLETE) ? true : false;
+            }
+        }
+
+        internal static int Encrypt(
+            SafeGssContextHandle context,
+            bool encrypt,
+            byte[] buffer,
+            int offset,
+            int count,
+            out byte[] outputBuffer)
+        {
+            outputBuffer = null;
+            Debug.Assert((buffer != null) && (buffer.Length > 0), "Invalid input buffer passed to Encrypt");
+            Debug.Assert((offset >= 0) && (offset < buffer.Length), "Invalid input offset passed to Encrypt");
+            Debug.Assert((count > 0) && (count <= (buffer.Length - offset)), "Invalid input count passed to Encrypt");
+
+            using (SafeGssBufferHandle outputToken = new SafeGssBufferHandle())
+            {
+                libgssapi.Status minorStatus;
+                libgssapi.Status status = libgssapi.GssWrap(out minorStatus, context, encrypt, buffer, offset, count, outputToken);
+                if (status != libgssapi.Status.GSS_S_COMPLETE)
+                {
+                    throw libgssapi.GssApiException.Create(SR.net_context_wrap_failed, status, minorStatus);
+                }
+
+                outputBuffer = new byte[outputToken.Length]; // Always return non-null
+                if (outputToken.Length > 0)
+                {
+                    Marshal.Copy(outputToken.Value, outputBuffer, 0, outputToken.Length);
```

Comment:
Nit: Replace outputToken.Length with outputBuffer.Length (see other places as well if this comment applies)